### PR TITLE
Release v9.0.0

### PR DESCRIPTION
**Pull request by jitu5**
## Description

# Release 9.0.0

## Major features and improvements

- Extending stateful URLs with node filters and expand/collapse modular pipelines. (#1799)
- Introduce `--include-hooks` option and remove `--ignore-plugins` from cli commands. (#1818)
- Add Dataset Factory Patterns to Experiment Tracking. (#1824)
- Add support for `JSONDataset` preview. (#1800)
- Upgrade to Node 18. (#1848)

## Bug fixes and other changes
- Increase Kedro-Viz timeout. (#1803)
- Remove demo data source and update feature hints. (#1804)
- Add markdown support for backticks in the pop-up reminder. (#1826)
- Fix posix path conversion on Windows in DatasetStatsHook. (#1843)
- Add `pydantic` pin to requirements. (#1861)
- Fix TRANSCODING_SEPARATOR import error. (#1866)
- Fix CircleCI build failure due to secret scan and missing toposort dependency. (#1819)
- Support Deferred Type Annotations in Dataset Previews. (#1798)  

## Community contributions

Many thanks to the following Kedroids for contributing PR to this release:
* [Pascal Brokmeier](https://github.com/pascalwhoop)

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by astrojuanlu — APPROVED** [review]
🚀 

Isn't there a place to acknowledge community contributors in the changelog, like we do in other plugins? In this case I think I only saw https://github.com/kedro-org/kedro-viz/pull/1809

**Review by stephkaiser — APPROVED** [review]

**Comment by jitu5** [comment]
> 🚀
> 
> Isn't there a place to acknowledge community contributors in the changelog, like we do in other plugins? In this case I think I only saw #1809

@astrojuanlu Sure I can section like this
Screenshot 2024-04-16 at 3 57 22 p m

**Review by SajidAlamQB — APPROVED** [review]
:shipit:

File: RELEASE.md [hunk 1]
```diff
@@ -23,6 +23,11 @@ Please follow the established format:
 - Fix posix path conversion on Windows in DatasetStatsHook. (#1843)
 - Add `pydantic` pin to requirements. (#1861)
 
+## Community contributions
+
+Many thanks to the following Kedroids for contributing PR to this release:
+* [Pascal Bro](https://github.com/pascalwhoop)
```
Comment (astrojuanlu) [inline RELEASE.md:29]:
```suggestion
* [Pascal Brokmeier](https://github.com/pascalwhoop)
```

File: src/components/update-reminder/update-reminder-content.js [hunk 2]
```diff
@@ -1,41 +1,27 @@
...
-      copy: '`%run_viz` line magic can be executed on jupyter notebooks with the options that Kedro-Viz supports on the command line',
-      buttonLink:
-        'https://docs.kedro.org/projects/kedro-viz/en/latest/kedro-viz_visualisation.html#running-kedro-viz-in-a-notebook',
-      buttonText: 'View the docs',
-    },
-    {
-      title: 'Kedro-Viz supports pydantic v2',
-      image: '',
-      copy: 'Kedro-Viz has migrated from pydantic v1 to v2',
+      copy: 'Now you can include `--include-hooks` options to all the viz cli commands and `%run_viz` Jupyter line magic',
```
Comment (rashidakanchwala) [inline src/components/update-reminder/update-reminder-content.js:17]:
```suggestion
      copy: To run hooks while running `kedro viz`, you must now include the `--include-hooks` option in the Viz CLI commands and the `%run_viz` Jupyter line magic. This change ensures hooks are executed only when explicitly requested, as they are no longer run by default in Kedro-Viz.
```

File: RELEASE.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ Please follow the established format:
 - Use present tense (e.g. 'Add new feature')
 - Include the ID number for the related PR (or PRs) in parentheses
 -->
-# Upcoming Release
+# Release 9.0.0
```
Comment (rashidakanchwala) [inline RELEASE.md:8]: Upgrade to Node 18 (#1848) not #1811

Comment (rashidakanchwala) [reply RELEASE.md:8]: and 'Add support for `JSONDatast` preview should be moved to major features and improvements, i think

Comment (rashidakanchwala) [reply RELEASE.md:8]: can we also add #1819, #1798,

Comment (jitu5) [reply RELEASE.md:8]: Sure, Let me add that.

**Review by Huongg — APPROVED** [review]
LGTM 👍 , thanks @jitu5 😄 
I've tested it locally and everything seems to work as expected to me

**Review by rashidakanchwala — COMMENTED** [review]
Amazing! Let's do it

**Review by rashidakanchwala — APPROVED** [review]

**Pull request merged**